### PR TITLE
[feat]: [BT-2309]: copyright check script added

### PR DESCRIPTION
**Pull request by abhayganvir**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40483)

### CODE REVIEW

**Comment by abhayganvir** [comment]
trigger copyrightcheck

**Comment by abhayganvir** [comment]
trigger copyrightcheck

**Comment by abhayganvir** [comment]
trigger copyrightcheck

File: scripts/jenkins/copyright-check.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Copyright 2022 Harness Inc. All rights reserved.
+# Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+# that can be found in the licenses directory at the root of this repository, also available at
+# https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+set -xe
+merge_summary=( $(git diff HEAD@{0} HEAD@{1} --name-only) )
+echo "Merge Summary: "${merge_summary[@]}
+error_data=()
+
+copyright_text="Copyright "$current_year" Harness Inc. All rights reserved."
```
Comment (USB-05) [inline scripts/jenkins/copyright-check.sh:11]: why are we using the current_year ? Is this the copyright year changes with every year ?

File: scripts/jenkins/copyright-check.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# Copyright 2022 Harness Inc. All rights reserved.
+# Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+# that can be found in the licenses directory at the root of this repository, also available at
+# https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+set -xe
+merge_summary=( $(git diff HEAD@{0} HEAD@{1} --name-only) )
```
Comment (USB-05) [inline scripts/jenkins/copyright-check.sh:7]: initialise this merge_summary same as error_data

**Comment by abhayganvir** [comment]
trigger copyrightcheck

**Review by USB-05 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40483#-:-NHcruDz0Y_X77CAwsko:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**